pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -112,6 +123,68 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .map(m -> {
+                               int indexOfSlash = reverseStringSearch('/', xpathExpressionArea.getText(), m.getInsertionEnd());
+                               String input = xpathExpressionArea.getText().substring(indexOfSlash + 1, m.getInsertionEnd());
+                               return Tuples.t(indexOfSlash, input);
+                           })
+                           .filter(t -> t._2.matches("[a-zA-Z]+"))
+                           .subscribe(e -> autoComplete(e._1, e._2));
+
+        xpathExpressionArea.addEventHandler(KeyEvent.KEY_PRESSED, t -> {
+            if (t.getCode().isLetterKey() || (t.isControlDown() && t.getCode().isWhitespaceKey()) || t.getCode() == KeyCode.BACK_SPACE) {
+                autoCompletePopup.show(xpathExpressionArea, 500, 500);
+            }
+
+        });
+
+        xpathExpressionArea.addEventHandler(KeyEvent.KEY_PRESSED, t -> {
+            if (t.getCode() == KeyCode.ESCAPE) {
+                autoCompletePopup.hide();
+            }
+        });
+
+
+    }
+
+    private void autoComplete(int slashPosition, String input) {
+        autoCompletePopup.getItems().clear();
```

Comment:
Instead of clearing here, and adding all at the end: don't clear here, and use setItems at the end. This will prevent some twitching